src/vm/analysis/cost_counter/natives.rs

```diff
@@ -0,0 +1,589 @@
+        Keccak256 => SimpleCostSpecification::new_diskless(Linear(constants::KECC256_A, constants::KECC256_B)),
+
+        ConsSome | ConsOkay | ConsError 
+                             => SimpleCostSpecification::new_diskless(Linear(constants::CONS_OPTION_A, constants::CONS_OPTION_B)),
+        DefaultTo            => SimpleCostSpecification::new_diskless(Linear(constants::DEFAULT_TO_A,  constants::DEFAULT_TO_B)),
+        Expects | ExpectsErr => SimpleCostSpecification::new_diskless(Linear(constants::EXPECTS_A,     constants::EXPECTS_B)),
+        Print =>                SimpleCostSpecification::new_diskless(Linear(constants::PRINT_A, constants::PRINT_B)),
+
+        IsOkay | IsNone => SimpleCostSpecification::new_diskless(Constant(constants::IS_OPTION_COST)),
+        AsContract =>      SimpleCostSpecification::new_diskless(Constant(constants::AS_CONTRACT_COST)),
+
+        // assert max len is linear right now due to clone.
+        AssertsMaxLen => SimpleCostSpecification::new_diskless(Linear(constants::LEN_CHECK_A, constants::LEN_CHECK_B)),
+        Concat => SimpleCostSpecification::new_diskless(Linear(constants::CONCAT_A, constants::CONCAT_B)),
+        Append => SimpleCostSpecification::new_diskless(Linear(constants::APPEND_A, constants::APPEND_B)),
+
+        // Fetches need to be handled specially.
+        //    read runtime costs are linear (may be super-linear to parse) in the _stored type_, not the
+        //      argument length
+        FetchVar => { return CostSpecification::Special(SpecialCostType::FetchVar) },
+        FetchEntry => { return CostSpecification::Special(SpecialCostType::FetchEntry) },
+        FetchContractEntry => { return CostSpecification::Special(SpecialCostType::FetchContractEntry) },
+        SetVar  => { return CostSpecification::Special(SpecialCostType::SetVar) },
+        SetEntry => { return CostSpecification::Special(SpecialCostType::SetEntry) },
+        InsertEntry => { return CostSpecification::Special(SpecialCostType::InsertEntry) },
+        DeleteEntry => { return CostSpecification::Special(SpecialCostType::DeleteEntry) },
+        GetBlockInfo => SimpleCostSpecification {
+            read_count: Constant(1),
+            read_length: Constant(constants::GET_BLOCK_INFO_READ_LEN),
+            write_length: Constant(0), write_count: Constant(0),
+            runtime: Constant(constants::GET_BLOCK_INFO_COST)
+        },
+        MintAsset => { return CostSpecification::Special(SpecialCostType::MintAsset) },
+        TransferAsset => { return CostSpecification::Special(SpecialCostType::TransferAsset) },
+        TransferToken => SimpleCostSpecification {
+            read_count: Constant(1), write_count: Constant(1),
+            read_length: Constant(constants::TRANSFER_TOKEN_READ_LEN),
+            write_length: Constant(constants::TRANSFER_TOKEN_WRITE_LEN),
+            runtime: Constant(constants::TRANSFER_TOKEN_COST),
+        },
+        MintToken => SimpleCostSpecification {
```

Comment:
When minting a some tokens, we are checking that we're not overflowing the total supply.
I'm not sure - is the cost of this check included there?

Reply:
Yes, it should be, though the constant number of reads might need to be adjusted -- MintToken will perform at least 1 read to get the current supply (though, I think the runtime implementation may perform 2 reads right now).